inc/Engine/Admin/Beacon/Beacon.php

```diff
@@ -84,12 +99,20 @@ public function insert_script() {
 	 */
 	private function get_user_locale() {
 		if ( isset( $this->locale ) ) {
-			return $this->locale;
+			/**
+			 * Filters the locale ID for Beacon
+			 *
+			 * @since 3.6
+			 *
+			 * @param string $locale The locale ID.
+			 */
+			return apply_filters( 'rocket_beacon_locale', $this->locale );
 		}
 
 		$this->locale = current( array_slice( explode( '_', get_user_locale() ), 0, 1 ) );
 
-		return $this->locale;
+		// This filter is documented in inc/Engine/Admin/Beacon/Beacon.php.
+		return apply_filters( 'rocket_beacon_locale', $this->locale );
```

Comment:
Suggestion:
```php
if ( ! isset( $this->locale ) ) {
	$this->locale = current( array_slice( explode( '_', get_user_locale() ), 0, 1 ) );
}
/**
 * Filters the locale ID for Beacon.
 *
 * @since 3.6
 *
 * @param string $locale The locale ID.
 */
return (string) apply_filters( 'rocket_beacon_locale', $this->locale );
```